tests/e2e/replication/show_while_creating_invalid_state.py

```diff
@@ -403,5 +440,1016 @@ def check_roles():
     assert actual_data == expected_data
 
 
+def test_async_replication_when_main_is_killed():
+    # Goal of the test is to check that when main is randomly killed:
+    # -the ASYNC replica always contains a valid subset of data of main.
+    # -the SYNC replica always contains the exact data that was in main.
+    # We run the test 20 times, it should never fail.
+
+    # 0/ Start main and replicas.
+    # 1/ Register replicas.
+    # 2/ Insert data in main, and randomly kill it.
+    # 3/ Check that the ASYNC replica has a valid subset.
+
+    for test_repetition in range(20):
+        # 0/
+        data_directory_main = tempfile.TemporaryDirectory()
+        data_directory_replica = tempfile.TemporaryDirectory()
+        CONFIGURATION = {
+            "async_replica": {
+                "args": ["--bolt-port", "7688", "--log-level=TRACE"],
+                "log_file": "async_replica.log",
+                "setup_queries": ["SET REPLICATION ROLE TO REPLICA WITH PORT 10001;"],
+                "data_directory": f"{data_directory_replica.name}",
+            },
+            "main": {
+                "args": ["--bolt-port", "7687", "--log-level=TRACE", "--storage-recover-on-startup=true"],
+                "log_file": "main.log",
+                "setup_queries": [],
+                "data_directory": f"{data_directory_main.name}",
+            },
+        }
+
+        interactive_mg_runner.start_all(CONFIGURATION)
+
+        # 1/
+        interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query(
+            "REGISTER REPLICA async_replica ASYNC TO '127.0.0.1:10001';"
+        )
+
+        # 2/
+        for index in range(50):
+            interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query(f"CREATE (p:Number {{name:{index}}})")
+            if random.randint(0, 100) > 95:
+                interactive_mg_runner.kill(CONFIGURATION, "main")
```

Comment:
Please consider logging when main was killed.

Reply:
Done. We will log if the test fail